pkg/frontend/openshiftcluster_putorpatch.go

```diff
@@ -176,6 +180,21 @@ func (f *frontend) _putOrPatchOpenShiftCluster(ctx context.Context, log *logrus.
 		// around LB/PLS bug
 		doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID = strings.ToLower(doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID)
 
+		const resourcegroupIDPatternText = `(?i)subscriptions/(.+)/resourceGroups/(.+)`
+		resourcegroupIDPattern := regexp.MustCompile(resourcegroupIDPatternText)
+		a := resourcegroupIDPattern.FindStringSubmatch(doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID)
+
+		fpAuthorizer, err := f.env.FPAuthorizer(f.env.TenantID(), f.env.Environment().ResourceManagerEndpoint)
+		if err != nil {
+			return nil, err
+		}
+
+		resourceGroupsClient := features.NewResourceGroupsClient(f.env.Environment(), a[1], fpAuthorizer)
+		rgexists, err := resourceGroupsClient.CheckExistence(ctx, a[2])
```

Comment:
Minor suggestion (IMHO):
"rgexists" looks like a name for a boolean. What about "resourceGroupExistenceResult" or something similar? 
I prefer long descriptive names rather than short ones (when it makes sense).